lib/persistence.js

```diff
@@ -162,18 +162,31 @@ Persistence.readHashAll = function(hash, callback) {
 };
 
 // Return the value associated with the key *key* (no associated hash)
-Persistence.readKey = function (key, callback) {
+Persistence.readKey = function (key, callback, expireTTL) {
+  var multi = Persistence.redis().multi();
+
   logging.debug('readKey:', key);
-  Persistence.redis().get(key, function (err, reply) {
-    if(err) throw new Error(err);
+  multi.get(key, function (err, reply) {
+    if (err) throw new Error(err);
+    if (expireTTL) {
+      multi.expire(key, expireTTL);
+    }
```

Comment:
This should be done outside the callback. Multi works like a queue of commands. So, add a get, then expire and then exec.

If you expire a key which is not there, nothing bad will happen (thanks to redis).